docs/proposal/2022-09-28-event-manager-extensibility.md

```diff
@@ -0,0 +1,302 @@
+you can see base gRPC server and client implementations in `grpc.go` which is shared among plugins. `interface.go` contains the actual API to expose to consumers. Finally,
+`plugin.go` is for plugin definition which is a simple struct for Hashicorp's Go plugin system.
+
+### Plugins Folder
+In this folder, you can see examples `kubectl` and `Kubernetes`. Each plugin has its own dedicated folders which are typically go module projects. 
+Each plugin folder contains a simple Go file which contains the actual business logic of plugin. You can also see a `Makefile` to manage the build
+process of that plugin. They might not be a simple Go program, that's why there is no unified `Makefile` that builds any Go project, it can contain
+customer specific build flows. For `kubectl` and `Kubernetes`, they are same for now. 
+
+### How to build plugins?
+`plugins` folder only contains the source code of the plugins, and consumer (Botkube or playground project on our case), needs the executable versions
+of those plugin to call via Hashicorp's plugin package. There is a Github Action for plugin release flow which you can see [here](https://github.com/huseyinbabal/Botkube-plugins/blob/main/.github/workflows/release.yaml). 
+In this flow, the only thing you need to do is pushing a new tag in a format `<plugin_name>/<version>` like `Kubernetes/v1.0.7`. This will build `Kubernetes` plugin, create a release with version
+`v1.0.7` and go binary executable will be attached to release artifacts. Notice that, those artifacts are used in consumer side to be used as parameter to Hashicorp's Go Plugin.
+
+### How can I know the metadata of plugins?
+In `Botkube-plugins` project, we have a plugin index file as you can also see it [here](https://github.com/huseyinbabal/Botkube-plugins/blob/main/index.json). That contains basic metadata of each plugin, and notice that they are managed manually
+right now, we can automate and do find/replace whenever there is a change to plugins. Here is an example;
+```json
+[
+  {
+    "name": "Kubernetes",
+    "type": "source",
+    "description": "Kubernetes source plugin to consume events from Kubernetes Events API.",
+    "version": "v1.0.9"
+  },
+  {
+    "name": "kubectl",
+    "type": "executor",
+    "description": "Kubectl executor plugin to handle given command in configured k8s cluster.",
+    "version": "v1.0.9"
+  }
+]
+```
+
+### How can I provide plugin configuration?
+If you use terraform before, they use a notation like `<pluginname>_...` to pass settings to specific plugin. So, in Botkube, if somebody provides `KUBERNETES_CONFIG_PATH=something`, the `kubernetes` plugin executable can be run as follows.
+```go
+client := plugin.NewClient(&plugin.ClientConfig{
+    Plugins:          m.pluginMap(metadata),
+    VersionedPlugins: nil,
+    Cmd:              exec.Command(metadata.Path, "--config", "something"), // Plugin specific params goes here
```

Comment:
IMO we should just add an option to pass a raw config via `ConfigYAML string` property. In this way, each plugin can easily unmarshal it to a given struct. Providing such configuration: https://github.com/kubeshop/botkube/blob/ef6d42a8da292503664e744fb0224dec1d83ce19/helm/botkube/values.yaml#L61-L170
via env variables will too complicated.

Reply:
Sure, during plugin initialization, we can pass it with the config file.

Reply:
I think it would be great to mention it in the doc, just to make sure we remember about it 👍

Reply:
updated accordingly